### DISCUSSION TITLE
Experiment with Examples in Documentation

### DISCUSSION THREAD

**Discussion by magnus-madsen** [flix/flix#9399] [Ideas]
For example:

```
    ///
    /// Returns `Some(x)` if `x` is the first element of `xs`.
    ///
    /// Returns `None` if `xs` is empty.
    ///
    /// # Examples
    /// ```
    /// head(Nil) == None
    /// ```
    ///
    /// ```
    /// head(1 :: 2 :: 3 :: Nil) == Some(1)
    /// ``` 
    pub def head(xs: List[a]): Option[a] = match xs {
        case Nil => None
        case x :: _ => Some(x)
    }
```

**Comment by mlutze** [comment]
Is this just a stylistic idea or are you thinking of something like [doctest](https://docs.python.org/3/library/doctest.html)?

**Comment by magnus-madsen** [comment]
I was thinking mostly for doc in VSCode, but we could make a principled design based on doctest or similar.

**Comment by magnus-madsen** [comment]
What should the syntax be and what should we do with it? Parse it and type check it? (And ultimately test it?)

**Comment by mlutze** [comment]
For now I think we would treat it similar to laws, so we parse and type check and then throw away.

For syntax, I am not sure. A few thoughts:
- should be simple so as not to disrupt normal reading of documentation
- should be able to include both tested and non-tested code blocks
